### ISSUE TITLE
Several tags in the last commit.

### ISSUE THREAD

**Issue by jidckii** [goreleaser/goreleaser#4209]
### What happened?

I've run into a limitation that if there are multiple tags in the last commit, the gorreleaser selects the wrong one that I want.
For example, I released prerelease v0.9.2-beta2 and then from that same commit v0.9.2. So in the second case, goreleaser again took the v0.9.2-beta2 tag, and not v0.9.2. What to do if a commit has multiple tags. Some of them may be service and not related to the dependence on Semantic Versioning.

### How can we reproduce this?

First create tag v0.9.2-beta2 , next create tag v0.9.2. In both cases, the first tag will be selected.

### goreleaser version

```bash
$ goreleaser -v
  ____       ____      _                            ____
 / ___| ___ |  _ \ ___| | ___  __ _ ___  ___ _ __  |  _ \ _ __ ___
| |  _ / _ \| |_) / _ \ |/ _ \/ _` / __|/ _ \ '__| | |_) | '__/ _ \
| |_| | (_) |  _ <  __/ |  __/ (_| \__ \  __/ |    |  __/| | | (_) |
 \____|\___/|_| \_\___|_|\___|\__,_|___/\___|_|    |_|   |_|  \___/
goreleaser-pro: Deliver Go Binaries as fast and easily as possible
https://goreleaser.com/pro

GitVersion:    1.19.2-pro
GitCommit:     f470f25cfbf7dd93acf7c7ab66defeb1bc2188de
GitTreeState:  clean
BuildDate:     2023-07-07T16:57:14Z
BuiltBy:       goreleaser
GoVersion:     go1.20.5
Compiler:      gc
ModuleSum:     unknown
Platform:      linux/amd64
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this

### Supporter

- [X] I am a [sponsor](https://github.com/sponsors/caarlos0/) or a [Pro](https://goreleaser.com/pro) customer

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caarlos0** [comment]
duplicate of https://github.com/goreleaser/goreleaser/issues/4134

**Comment by caarlos0** [comment]
tldr: https://goreleaser.com/customization/git/

change `tag_sort` to `-version:creatordate` should do the trick if you always want the latest

**Comment by dannykopping** [comment]
Is `-version:creatordate` supposed to choose the latest version? The [docs](https://goreleaser.com/customization/git/) are pretty light on detail.

**Comment by caarlos0** [comment]
Hi!

these should be clearer indeed, check `man git-tag` and search for sort in there

will also update docs
